derive/src/utils.rs

```diff
@@ -45,10 +45,10 @@ pub fn const_eval_check_variant_indexes(
 	let mut recurse_indices = vec![];
 	for (ident, index) in recurse_variant_indices {
 		let ident_str = ident.to_string();
-		// We convert to u8 same as in the generated code.
+		// We convert to usize same as in the generated code.
 		recurse_indices.push(quote_spanned! { ident.span() =>
 			(
-				(#index) as ::core::primitive::u8,
+				(#index) as ::core::primitive::usize,
```

Comment:
in some very rare case it could still trigger truncation if type is like `repr
(u128)` and he is using u128 discriminant, doesn't harm to also have the flag here: `cast_possible_truncation`